app/services/google_auth_service.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class GoogleAuthService < AuthServiceBase
+  class << self
+    def find_user_by_token(google_id_token)
+      super
+    rescue Google::Auth::IDTokens::ExpiredTokenError => e
+      raise self::FailedAuthentication, e.message
+    end
+
+    private
+
+    def service_name
+      "google"
+    end
```

Comment:
Is this necessary? Looks like the inherited method might work.

Reply:
Oh, yes, it will work now